app/src/main/java/com/android/voyageur/model/trip/TripRepositoryFirebase.kt

```diff
@@ -64,6 +64,34 @@ class TripRepositoryFirebase(private val db: FirebaseFirestore) : TripRepository
         }
   }
 
+  /**
+   * Fetches trips that are discoverable and not created by the user.
+   *
+   * @param userId The ID of the user.
+   * @param onSuccess The callback to be invoked when the trips are fetched successfully.
+   * @param onFailure The callback to be invoked when an error occurs.
+   */
```

Comment:
The function includes clear Javadoc-style comments, describing its purpose and parameters. Thank you!